packages/compass-e2e-tests/tests/assistant.test.ts

```diff
@@ -0,0 +1,581 @@
+        before(async function () {
+          await setAIOptIn(browser, true);
+
+          mockAssistantServer.setResponse({
+            status: 200,
+            body: 'You should review the connection string.',
+          });
+        });
+
+        it('opens assistant with error message view prompt when clicking "Debug for me"', async function () {
+          await browser.connectWithConnectionString(
+            'mongodb-invalid://localhost:27017',
+            { connectionStatus: 'failure' }
+          );
+          await useErrorViewEntryPoint(browser);
+
+          const messages = await getDisplayedMessages(browser);
+          expect(messages).deep.equal([
+            {
+              text: 'Diagnose why my Compass connection is failing and help me debug it.',
+              role: 'user',
+            },
+            {
+              text: 'You should review the connection string.',
+              role: 'assistant',
+            },
+          ]);
+
+          expect(mockAssistantServer.getRequests()).to.have.lengthOf(1);
+        });
+      });
+    });
+  });
+});
+
+async function setAIFeatures(browser: CompassBrowser, newValue: boolean) {
+  await browser.openSettingsModal('ai');
+
+  // Wait for AI settings content to be visible
+  const aiSettingsContent = browser.$(
+    Selectors.ArtificialIntelligenceSettingsContent
+  );
+  await aiSettingsContent.waitForDisplayed();
+
+  const currentValue =
+    (await browser
+      .$(Selectors.SettingsInputElement('enableGenAIFeatures'))
+      .getAttribute('aria-checked')) === 'true';
+
+  if (currentValue !== newValue) {
+    await browser.clickParent(
+      Selectors.SettingsInputElement('enableGenAIFeatures')
+    );
+    await browser.clickVisible(Selectors.SaveSettingsButton);
+  }
+
+  const closeButton = browser.$(Selectors.CloseSettingsModalButton);
+  await closeButton.waitForClickable();
+  await closeButton.click();
+  await closeButton.waitForDisplayed({
+    reverse: true,
+  });
+}
+
+async function setAIOptIn(browser: CompassBrowser, enabled: boolean) {
+  // Reset the opt-in preference by using the execute command
+  await browser.setFeature('optInGenAIFeatures', enabled);
+
+  // Wait for the IPC to be processed
+  await browser.pause(500);
+}
+
+async function openAssistantDrawer(browser: CompassBrowser) {
+  const drawerButton = browser.$(Selectors.AssistantDrawerButton);
+  await drawerButton.waitForDisplayed();
+  await drawerButton.waitForClickable();
+  await drawerButton.click();
+}
+
+async function clearChat(browser: CompassBrowser) {
+  const clearChatButton = browser.$(Selectors.AssistantClearChatButton);
+  if (await clearChatButton.isDisplayed()) {
+    await clearChatButton.click();
+    const confirmButton = browser.$(
+      Selectors.AssistantConfirmClearChatModalConfirmButton
+    );
+    await confirmButton.waitForClickable();
+    await confirmButton.click();
+  }
+}
+
+async function getDisplayedMessages(browser: CompassBrowser) {
```

Comment:
I originally defined these functions in hooks but it was making the code too dirty so I moved them out here and made it pass browser instead... I imagine if chat becomes popular enough we can extend browser itself.

Reply:
That's fine - I usually add things locally and then only factor them out once necessary. I think that's a good way to work.